operator/internal/values/translation/translation.go

```diff
@@ -409,3 +446,62 @@ func GetConfigAsCode(c *platform.ConfigAsCodeSpec) *ValuesBuilder {
 	}
 	return &sv
 }
+
+// GetDeploymentDefaults returns the default nodeSelector and tolerations for all Deployments.
+// If pinToNodes is set, it expands it to the default nodeSelector and tolerations.
```

Comment:
Maybe add a note that correctness of this depends on the assertions checked by the extension?

Reply:
Actually I'm starting to think that the extension is overengineered.

Ok, reconcile fails faster with it, but otherwise it just adds complexity. I'd get rid of it and return an error here.